src/test/java/nextstep/subway/section/SectionAcceptanceTest.java

```diff
@@ -0,0 +1,146 @@
+		SectionRequest sectionRequest = new SectionRequest(gunDaeStationId, gangNamStationId, 2);
+		ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(sectionRequest);
+		// then
+		지하철_노선에_구간_포함됨(response, Arrays.asList("성수역", "건대입구역", "강남역", "사당역"));
+	}
+
+	@DisplayName("노선에 구간을 중간에 등록한다.역 - 기존 상행역 새로운 하행역")
+	@Test
+	void 노선에_구간_중간에_등록한다_기존_상역_새로운_하행역() {
+		// when
+		Long gangByunStationId = StationAcceptanceTest.지하철역_생성되어_있음(new StationRequest("강변역"));
+		SectionRequest sectionRequest = new SectionRequest(sungSuStationId, gangByunStationId, 2);
+		ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(sectionRequest);
+		// then
+		지하철_노선에_구간_포함됨(response, Arrays.asList("성수역", "강변역", "강남역", "사당역"));
+	}
+
+	@DisplayName("새로운 역을 등록할 경우 기존 역 사이 길이보다 크거나 같으면 등록을 할 수 없다.")
+	@Test
+	void 노선에_구간_중간에_등록한다_기존_역_사이_길이보다_크거나_같으면_등록을_할_수_없다() {
+		// when
+		Long gangByunStationId = StationAcceptanceTest.지하철역_생성되어_있음(new StationRequest("강변역"));
+		SectionRequest sectionRequest = new SectionRequest(sungSuStationId, gangByunStationId, 15);
+		ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(sectionRequest);
+		// then
+		지하철_노선에_구간_등록_실패(response);
+	}
+
+	@DisplayName("상행역과 하행역 둘 중 하나도 포함되어있지 않으면 추가할 수 없다")
+	@Test
+	void 상행역과_하행역_둘_중_하나도_포함되어있지_않으면_추가할_수_없다() {
+		// when
+		Long gangByunStationId = StationAcceptanceTest.지하철역_생성되어_있음(new StationRequest("강변역"));
+		Long gunDaeStationId = StationAcceptanceTest.지하철역_생성되어_있음(new StationRequest("건대입구역"));
+		SectionRequest sectionRequest = new SectionRequest(gunDaeStationId, gangByunStationId, 15);
+		ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(sectionRequest);
+		// then
+		지하철_노선에_구간_등록_실패(response);
+	}
+
+	@DisplayName("상행역과 하행역이 이미 노선에 모두 등록되어 추가할 수 없음")
+	@Test
+	void 상행역과_하행역이_이미_노선에_모두_등록되어_추가할_수_없음() {
+		// when
+		SectionRequest sectionRequest = new SectionRequest(sungSuStationId, gangNamStationId, 10);
+		ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(sectionRequest);
+		// then
+		지하철_노선에_구간_등록_실패(response);
+	}
+
+
+	private void 지하철_노선에_구간_등록_실패(ExtractableResponse<Response> response) {
+		assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+	}
+
+	ExtractableResponse<Response> 지하철_노선에_지하철역_등록_요청(SectionRequest sectionRequest) {
+		return RestAssured.given().log().all()
+			.body(sectionRequest)
+			.contentType(MediaType.APPLICATION_JSON_VALUE)
+			.when()
+			.post("/lines/" + lineNumber2Id + "/sections")
+			.then().log().all()
+			.extract();
+	}
+
+	public void 지하철_노선에_구간_포함됨(ExtractableResponse<Response> response, List<String> expectedStations) {
+		List<Station> stations = response.body().jsonPath().getList("stations", Station.class);
+		List<String> actualStations = stations.stream().map(station -> station.getName()).collect(Collectors.toList());
+		for (int index = 0; index < actualStations.size(); index++) {
+			assertThat(actualStations.get(index)).isEqualTo(expectedStations.get(index));
+		}
```

Comment:
`for문` 보다는 아래처럼 검증해보면 좋겠네요.

```suggestion
		assertThat(actualStations).containsAll(expectedStations);
```